flow/endpoint.go

```diff
@@ -14,6 +14,7 @@ type endpointOptions struct {
 	namespace string
 	podName   string
 	labels    []string
+	workloads map[string]string
```

Comment:
Any reason we don't just have a slice of `struct { Kind string, Name string }`?

Reply:
None other than convenience of `WithEndpointWorkloads` callers, but I guess we could do the `workloadsFromMap` dance there, what do you think?